pages/api/account/manage/repos.js

```diff
@@ -8,15 +8,52 @@ import Profile from "@models/Profile";
 export default async function handler(req, res) {
   const session = await getServerSession(req, res, authOptions);
 
-  if (req.method !== "GET") {
+  if(!["GET","PATCH"].includes(req.method)){
     return res.status(400).json({ error: "Invalid request: GET required" });
   }
 
-  const data = await getReposApi(session.username);
+  const username = session.username;
+  let data = [];
+  if(req.method === "GET"){
+    data = await getReposApi(username);
+  }
+  //const data = await getReposApi(username);
+  if(req.method === "PATCH"){
+    data = await updateReposOrderApi(username, req.body);
+  }
 
   return res.status(200).json(data);
 }
 
+export async function updateReposOrderApi(username, data) {
+  await connectMongo();
+  const log = logger.child({ username });
+
+  const repoList = await data.map(async (repo, idx) => {
+    try {
+      return await Profile.findOneAndUpdate(
+        {
+          username,
+          "repos._id": repo._id,
+        },
+        {
+          $set: {
+            "repos.$.order": idx,
+          },
+        },
+      );
+    } catch (e) {
+      log.error(e, `failed to update repo order for username: ${username}`);
+    }
+  });
+
+  const repos = await Promise.allSettled(repoList).then(() => { 
+    return getReposApi(username);
+  })
```

Comment:
Is this needed? Because the map and content are already using `await`

Reply:
Actually this yes is needed because, if we don't wait for all the promises to be settled, the reorder happens, but when you save the returned order is the same it had before, because it hasn't finished saving yet. If you refresh the page you actually see the correct order though. 
In this way it shows you the right order as you save.
Don't know if I'm missing some node points her  or maybe on server works, but locally it doesn't do the trick without specifying to wait all the ordering promises to be settled

Reply:
Ok, so I think if we need the promise all, then the map doesn't need the `await`? 

https://github.com/EddieHubCommunity/BioDrop/pull/9912/files/6082f0485096f7d9a8c4a00d422c4bc1058666ad#diff-0cf4a906bb37e0a3ff6060fb19d418de9313848bbbcf750c0effc667750bec9dR32

```diff
- const repoList = await data.map(async (repo, idx) => {
+ const repoList = data.map(async (repo, idx) => {
```

and possibly here also?

```
return await Profile.findOneAndUpdate(
```

Reply:
Yes indeed those weren't needed after all, removed them and it is still working! Thank you for pointing that out!